docs/Coding-Guidelines.md

```diff
@@ -1,34 +1,49 @@
-#### General
+# General
 
-Project Reunion prefers using industry-standard coding styles, guidelines, and patterns for any languages used in implementation or testing.
+Project Reunion prefers using industry-standard coding styles, guidelines, and patterns for any
+languages used in implementation or testing.
 
-#### C++
+## Languages
 
-Project Reunion implementations prefer C++ and C++/WinRT implementations.   New code must adhere to the **cppcoreguidelines** at
-https://github.com/isocpp/CppCoreGuidelines and be /W4 clean (for Visual C++.)
+This list is non-exhaustive; new guidelines for languages will be added over time.
 
-Prefer existing std:: and gsl:: types, but use **wil** (https://github.com/Microsoft/wil) types for Windows-specific helpers and lifecycle management
-rather than creating your own.
+### C++
 
-##### Catching Exceptions and HRESULT
+**DO** implement types using C++ and [C++/WinRT](https://github.com/microsoft/cppwinrt)
 
-Don't just catch `winrt::hresult_error` or other variations as that doesn't catch `std::bad_alloc`.
+**DO** follow the [CppCoreGudelines](https://github.com/isocpp/CppCoreGuidelines) for all new code.
 
-Use the following snippet to catch exceptions and retrieve their HRESULT:
+**DO** enable `/W4 /Wx` for all new code.
+
+**CONSIDER** using the [Windows Implementation Library](https://github.com/Microsoft/wil) for
+Windows-specific helpers rather than creating your own.
+
+**DO** use 4-space indentation instead of tab characters.
+
+##### Catching C++/WinRT Exceptions and HRESULT
+
+Exceptions should not be used for standard flow control.
```

Comment:
This may not be the place to have this discussion but can we recommend failfast (possibly via "noexcept" annotations) for most implementations? Having parameter validation with helpful RoOriginateError things at the boundary is fine, but in the bowels of an implementation I never want to see our own bugs manifest to the user as errors, we should just failfast.

Reply:
I get why you're asking for that, but I'm uncomfortable mandating or even recommending that globally for our components here yet.  Something to bring up in a future API design group meeting.